pkg/topology/kademlia/kademlia.go

```diff
@@ -118,17 +122,20 @@ func New(
 	logger logging.Logger,
 	o Options,
 ) (*Kad, error) {
-	if o.SaturationFunc == nil {
-		os := overSaturationPeers
-		if o.BootnodeMode {
-			os = bootNodeOverSaturationPeers
-		}
-		o.SaturationFunc = binSaturated(os, isStaticPeer(o.StaticNodes))
+
+	if o.OverSaturationCalc == nil {
+		o.OverSaturationCalc = defaultOverSaturationCalc
 	}
+
+	o.SaturationFunc = binSaturated(o.OverSaturationCalc, isStaticPeer(o.StaticNodes))
+
 	if o.BitSuffixLength == 0 {
 		o.BitSuffixLength = defaultBitSuffixLength
 	}
 
+	for i := 0; i < int(swarm.MaxBins); i++ {
+	}
+
```

Comment:
An empty loop?  To what purpose?

Reply:
hence the 'in progress' label :) thank you!